pkg/contract-manager/reservation_controller.go

```diff
@@ -47,13 +55,208 @@ type ReservationReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/reconcile
 func (r *ReservationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	klog.Info("Reconciling Reservation")
+	log := ctrl.LoggerFrom(ctx, "discovery", req.NamespacedName)
+	ctx = ctrl.LoggerInto(ctx, log)
 
-	// TODO(user): your logic here
+	// var contract *reservationv1alpha1.Contract
+
+	var reservation reservationv1alpha1.Reservation
+	if err := r.Get(ctx, req.NamespacedName, &reservation); client.IgnoreNotFound(err) != nil {
+		klog.Errorf("Error when getting Reservation %s before reconcile: %s", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	} else if err != nil {
+		klog.Infof("Reservation %s not found, probably deleted", req.NamespacedName)
+		return ctrl.Result{}, nil
+	}
+
+	if reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseSolved &&
+		reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseTimeout &&
+		reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseFailed &&
+		reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseRunning &&
+		reservation.Status.Phase.Phase != nodecorev1alpha1.PhaseIdle {
+
+		klog.Infof("Reservation %s started", reservation.Name)
+		setReservationPhase(&reservation, nodecorev1alpha1.PhaseRunning)
+
+		if err := r.updateReservationStatus(ctx, &reservation); err != nil {
+			klog.Errorf("Error when updating Reservation %s status before reconcile: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	switch reservation.Status.Phase.Phase {
+	case nodecorev1alpha1.PhaseRunning:
+		klog.Infof("Reservation %s running", reservation.Name)
+		if reservation.Status.Phase.Phase == nodecorev1alpha1.PhaseSolved {
+			return ctrl.Result{}, nil
+		}
+		// Reserve the resources with a certain FlavourID
+		transaction, err := reserveFlavour(ctx, &reservation)
```

Comment:
A transaction is something that is related JUST for the REAR Protocol. Do u think it can be useful to create a CRD for that? Remember that I save in cache all the transactions (as structs).

Reply:
Yes, i think that. Just because we need to save the transactions in a sort of DB. If the component crashes we lose them. Transactions are strictly related to contracts, are something related to payments in the future. Do not you think that is an important info?

Reply:
Fixed, created a Transaction Resource, used a function `forgeTransaction` that create a CR based on the CRD `Transaction` Object.